### ISSUE TITLE
tft.scale_by_min_max always calculates 0.0 as min and max values

### ISSUE THREAD

**Issue by baconmaniac** [tensorflow/transform#228]
Hello,

This is my first Github ticket ever, please be patient if I do things wrong, I'm trying my best :-)

I'm trying to scale my data using the tft.scale_by_min_max function that comes with Tensorflow-Transform. I noticed that the results differ quite heavily from my own calculations (I originally tried to verify my own code using Tensorflow-Transform). After some buzzing around (debugging those functions without eager execution is a pain but maybe I'm also just a fool haha) I managed to locate the error in the call of _numeric_combine in analyzers.py's _min_and_max function. It seems to always return 0 regardless of the min and max values passed to it.

Im using tensorflow 2.3.0 and transform 0.28.0.

For reproduction:

```
import tensorflow as tf
import tensorflow_transform as tft

A = tf.constant([2.0, -2.0, 2.0])
min = tf.reduce_min(A)
max = tf.reduce_max(A)

# Scale between -1 and 1
A_scaled = -1 + (tf.math.subtract(A, min)*2)/(max-min) # Results in [1, -1, 1] as expected

A_scaled_transform = tft.scale_by_min_max(A, output_min=-1.0, output_max=1.0) # Results in [ 0.7615942, -0.7615942,  0.7615942]
```
Let me know if you need any additional information.

Thank you and have a nice day!

**Comment by zoyahav** [comment]
Trying to run this in a preprocessing_fn I'm also getting [1, -1, 1], could you please share your entire repro?
(The one shared above wouldn't work because tft.scale_by_min_max must be wrapped in a preprocessing_fn, otherwise you'd get a placeholder related error)
See for reference: https://github.com/tensorflow/transform/blob/master/examples/simple_example.py

A couple of notes here:
TFT mappers such as tft.scale_by_min_max are supposed to be applied to tensors which are the inputs to your preprocessing_fn, this way the min/max are computed over the entire dataset, and then applied to the entire dataset.
What's happening in the repro above is computing min/max over a single tensor, this is not the intended use case for this.

Also, the results you're seeing are the values of sigmoid(A). tft.scale_by_min_max applied the sigmoid function when the analysis dataset is empty. This also indicates that there's likely an issue with how the TFT pipeline is constructed. Documented here:
https://www.tensorflow.org/tfx/transform/api_docs/python/tft/scale_by_min_max#returns

**Comment by arghyaganguly** [comment]
Closing this due to inactivity.Please feel free to reopen with the required information asked by @zoyahav.
